### ISSUE TITLE
Uri parser is broken

### ISSUE THREAD

**Issue by rossabaker** [http4s/http4s#320]
```scala
scala> Uri.fromString("http://45.55.173.208:5000/api/v1/authenticate").getOrElse(null)
res8: org.http4s.Uri = http://45.55.173.20
```

@kryptt points out it happens if the last octet >= 200.

**Comment by bryce-anderson** [comment]
Specifically, the IPv4 portion of our parser is broken. A PR is waiting for review.

**Comment by rossabaker** [comment]
Fix will be released in v0.8.3.
